index.js

```diff
@@ -54,6 +67,11 @@ function number(start, end, size){
  */
 
 function duration(start, end, duration){
+  var days = DAYS[duration];
+  if(days) {
```

Comment:
Not sure about the `floor` / `ceil` stuff below but this seemed simpler for larger chunks which can be defined in days.